Sources/_StringProcessing/Compiler.swift

```diff
@@ -155,6 +155,31 @@ class Compiler {
       throw unsupported(node.renderAsCanonical())
     }
   }
+  
+  func emitAny() throws {
+    switch (options.semanticLevel, options.dotMatchesNewline) {
+    case (.graphemeCluster, true):
+      builder.buildAdvance(1)
+    case (.graphemeCluster, false):
+      builder.buildConsume { input, bounds in
+        input[bounds.lowerBound].isNewline
+          ? nil
+          : input.index(after: bounds.lowerBound)
+      }
+
+    case (.unicodeScalar, true):
+      // TODO: builder.buildAdvanceUnicodeScalar(1)
+      builder.buildConsume { input, bounds in
+        input.unicodeScalars.index(after: bounds.lowerBound)
+      }
+    case (.unicodeScalar, false):
+      builder.buildConsume { input, bounds in
+        input[bounds.lowerBound].isNewline
+          ? nil
+          : input.unicodeScalars.index(after: bounds.lowerBound)
+      }
+    }
+  }
```

Comment:
Do we have tests for the combination of scalar-semantic with different newline behaviors?